Src/EmbeddedScripts.CSharp.Roslyn.Compilation/CompiledCodeRunner.cs

```diff
@@ -9,15 +9,15 @@ public class CompiledCodeRunner : ICodeRunner
         private Container _container = new();
         private CodeGeneratorForCompilation _codeGenerator = new();
 
-        public Task RunAsync(string code)
+        public Task<ICodeRunner> RunAsync(string code)
         {
             var generatedCode = _codeGenerator.GenerateCode(code, _container);
 
             var compilation = new CodeCompiler(generatedCode, _container).Compilation;
             var instance = new InstanceCreator(compilation).CreateInstanceOf(_codeGenerator.ClassName, _container);
             instance.InvokeMethod(_codeGenerator.MethodName);
             
-            return Task.CompletedTask;
+            return Task.FromResult(this as ICodeRunner);
```

Comment:
What happens if the code we compile is async? Will we wait here until it's executed to completion?

Reply:
If we have an `await` statements in code, then runner will throw `ScriptSyntaxErrorException` because template I use for code generation defines synchronous method. Otherwise it will continue code execution.